app/models/ext_management_system.rb

```diff
@@ -503,4 +503,38 @@ def blacklisted_event_names
   def self.blacklisted_events
     BlacklistedEvent.where(:provider_model => name, :ems_id => nil)
   end
+
+  def decrypt_passwords(creds)
```

Comment:
This should be a private method...I'm also concerned on the method naming, that it's too generic, but I can't think of something better.